### ISSUE TITLE
 UserVoiceStateUpdated Event not fired when User initially connects to voice channel

### ISSUE THREAD

**Issue by GeorgZipp** [RogueException/Discord.Net#956]
I'm trying to have my bot recognize when any guild member joins a specific voice channel and then react to it. I tried to do this by subscribing to the **UserVoiceStateUpdated** event. This basically works as expected, however when a user first joins any voice channel, the event is not fired. When the user then switches from one voice channel to another, it works fine. As soon as you completely disconnect the user from voice and reconnect it once again requires changing the voice channel before my event handler is executed.

I'm not doing anything fancy, just subscribing to the event like this
`_client.UserVoiceStateUpdated += HandleUserVoiceStateUpdated;`

HandleUserVoiceStateUpdated can literally just be an empty method.

----------

Just to clarify again, this is what I'm doing and how the bot reacts to it:

 1. User connects to voice - UserVoiceStateUpdated is not fired
 2. User then switches to another voice channel - UserVoiceStateUpdated triggers my event handler correctly
 3. User switches voice channel once again - UserVoiceStateUpdated triggers again
 4. User disconnects from voice completely - UserVoiceStateUpdated triggers
 5. User reconnects to any voice channel - UserVoiceStateUpdated handler is not executed
 6. User switches voice channels - UserVoiceStateUpdated triggers as expected

----------

Tried with both a .net Framework 4.7 and .net Core 2.0, as well as Discord.net versions 1.0, 1.0.1 and 1.0.2. Always the same.

**Comment by SubZero0** [comment]
Couldn't repro your issue.
Simple bot using .Net Core 2.0 and Discord.Net (1.0.2 and 2.0-880).
```cs
_client.UserVoiceStateUpdated += (user, before, after) =>
{
    Console.WriteLine($"VoiceStateUpdate: {user} - {before.VoiceChannel?.Name ?? "null"} -> {after.VoiceChannel?.Name ?? "null"}");
    return Task.CompletedTask;
};
```

I did exactly what you listed:

2.0-880
![image](https://user-images.githubusercontent.com/3085008/36275133-20b54c98-1271-11e8-88c3-a0e1ef2e2193.png)

1.0.2
![image](https://user-images.githubusercontent.com/3085008/36275168-395c52d2-1271-11e8-933b-7f962c543abe.png)

**Comment by GeorgZipp** [comment]
First of all, thanks for investigating.

After I read your post I added more logging to my simple test program and found this: "A UserVoiceStateUpdated handler is blocking the gateway task." That message is logged by Discord.net every time I would usually expect the event to trigger.

I made sure to kill all processes but it was still happening. So I started completely from scratch, created a new discord app, new bot users, etc... and it worked! The conclusion I was left with is that I must have made some kind of configuration error. Sorry for wasting your time. ;)
